### DISCUSSION TITLE
File Reader get response for HL7?

### DISCUSSION THREAD

**Discussion by ppazos** [nextgenhealthcare/connect#5240] [General]
Hi, checking a channel for a client I found they are reading HL7 v2.x as files with a File Reader and that the response is auto-generated, so I tested this kind of configuration and in the channel logs it seem it generates a response but I'm not sure how the client (upstream filesystem) can receive the response, since there is no file generated for the ACK.

So, how should this work? I mean a File Reader reading HL7 v2.x messages as files and auto-generating the response. How can the response be accessed? Where is Mirth actually putting that response that appears in the logs?

Thanks!

**Comment by rjkroll** [comment]
Well, there is no way the upstream client has access to the response of a File Reader unless you send it to them.    You can access the Response variable/transformer of any channel pretty easily.    You could then pass the response to another channel, where it sends it back to the upstream client or does something else with it.

**Comment by ppazos** [comment]
Thanks @rjkroll 

"....you can access the Response variable/transformer of any channel pretty easily. "

How? Can you expand?

**Comment by rjkroll** [comment]
In your Destination Response Transformer, you can see what response you are getting/producing by having a Response Transformer Step that does the following:

var rMsg = response.getMessage();
logger.info(rMsg);

The, if you wanted to send this response to another channel(which then sends it to an upstream client), you could use something like this:

if (responseStatus == SENT) {
    router.routeMessageByChannelId('channel ID here', response.getMessage());
}

**Comment by tonygermano** [comment]
The auto-generated ACK is created by the source connector. You can use the [ACKGenerator](http://javadocs.mirthcorp.com/connect/4.0.1/user-api/com/mirth/connect/server/userutil/ACKGenerator.html) from the User API to create the same ACK in a second File Writer destination. There's probably no reason to have the auto-generated response turned on since it doesn't go anywhere for a File Reader.
